src/engine-actions/translate-swap.js

```diff
@@ -88,10 +88,15 @@ async function translateSwap (takerAddress, swapHash, amount, extendedTimeLockDe
 
   this.logger.debug('Calling queryRoutes on lnd w/ params', { params: queryRoutesReq })
 
-  const [ { routes = [] }, { blockHeight = '' } ] = await Promise.all([
-    queryRoutes(queryRoutesReq, { client: this.client }),
-    getInfo({ client: this.client })
-  ])
+  try {
+    var [ { routes = [] }, { blockHeight = '' } ] = await Promise.all([
+      queryRoutes(queryRoutesReq, { client: this.client }),
+      getInfo({ client: this.client })
+    ])
+  } catch (e) {
+    this.logger.error(e)
+    return { permanentError: e.message }
```

Comment:
How can you be certain that the error returned is a permanent error? This could have a side effect of making the swap non-atomic in certain cases.

Reply:
@treygriffith I think in this case, since we haven't started translation (only trying to find a route), that we are OK to send a permanent failure. FWIW, we return a permanent error is `routes = 0`, so if this call fails, it is technically the same thing.

Reply:
Yes, I agree, since this is just pulling information and not initiating a payment, it is ok to do a permanent error here.